### PR TITLE
{Network} `az network nic`: Migrate commands to codegen v2

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

az network nic

**Description**

Total: 8 commands

**Testing Guide**

**History Notes**

[Network] BREAKING CHANGE: `az network nic create/update`: Rename output property `enableIpForwarding` to `enableIPForwarding` to keep consistent with the name in API
[Network] BREAKING CHANGE: `az network nic create/update`: Rename output property `privateIpAllocationMethod` to `privateIPAllocationMethod` to keep consistent with the name in API
[Network] BREAKING CHANGE: `az network nic create/update`: Rename output property `publicIpAddress` to `publicIPAddress` to keep consistent with the name in API

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -4181,82 +4182,191 @@ def create_cross_region_load_balancer(cmd, load_balancer_name, resource_group_na
...
+        args = self.ctx.args
+        if has_value(args.network_security_group):
+            args.nsg.id = args.network_security_group
+        if has_value(args.edge_zone):
+            args.extended_location.name = args.edge_zone
+            args.extended_location.type = "EdgeZone"
+        ip_configuration = {
+            "name": "ipconfig1",
+            "private_ip_address": args.private_ip_address,
+            "private_ip_address_version": args.private_ip_address_version,
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:4312]: Do we need to set `private_ip_address_version` by default value which is `IPV4`, when private_ip_address is not provided
```suggestion
            "private_ip_address_version": args.private_ip_address_version if has_value(args.private_ip_address),
```

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -4181,82 +4182,191 @@ def create_cross_region_load_balancer(cmd, load_balancer_name, resource_group_na
...
+            "load_balancer_backend_address_pools": [{"id": x} for x in args.lb_address_pools] if has_value(args.lb_address_pools) else None,
+            "load_balancer_inbound_nat_rules": [{"id": x} for x in args.lb_inbound_nat_rules] if has_value(args.lb_inbound_nat_rules) else None,
+        }
+        args.ip_configurations = [ip_configuration]
 
-    if public_ip_address:
-        ip_config.public_ip_address = PublicIPAddress(id=public_ip_address)
-    nic.ip_configurations = [ip_config]
+    def _output(self, *args, **kwargs):
+        result = self.deserialize_output(self.ctx.vars.instance, client_flatten=True)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:4324]:
```suggestion
        result = super()._output(*arg, **kwargs)
```

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 1 above)
Comment (necusjz) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:4312]: Correct.

**Review by kairu-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**